internal/datanode/data_node.go

```diff
@@ -1003,22 +1048,21 @@ func (node *DataNode) AddSegment(ctx context.Context, req *datapb.AddSegmentRequ
 			}, nil
 		}
 	}
-	// Update # of rows of the given segment.
-	ds.replica.updateStatistics(req.GetSegmentId(), req.GetRowNum())
+	ds.flushingSegCache.Remove(req.GetSegmentId())
 	return &commonpb.Status{
 		ErrorCode: commonpb.ErrorCode_Success,
 	}, nil
 }
 
 func importFlushReqFunc(node *DataNode, req *datapb.ImportTaskRequest, res *rootcoordpb.ImportResult, schema *schemapb.CollectionSchema, ts Timestamp) importutil.ImportFlushFunc {
```

Comment:
shouldn't a function in datanode

Reply:
and I don't like the coding style that each function go across 250 lines.
split it into multiple functions

Reply:
Splited.